tincr/io/design/rapidsmith.tcl

```diff
@@ -0,0 +1,238 @@
+package provide tincr.io.design 0.0
+package require Tcl 8.5
+package require tincr.cad.design 0.0
+package require tincr.cad.device 0.0
+package require tincr.cad.util 0.0
+
+namespace eval ::tincr:: {
+    namespace export \
+        run_rapidsmith_command
+}
+
+## Executes a command send from a running instance of RapidSmith. The "rs_start" and "rs_end"
+#   are used to communicate with RapidSmith when the command has started or completed.
+#
+proc ::tincr::run_rapidsmith_command { cmd } {
+    puts "rs_start"
+    {*}$cmd
+    puts "rs_end"
+} 
+
+## Tests if the specified bel is a routethrough under a certain configuration. Specifically, 
+#   this function checks that [get_property CONFIG.EQN]="(expected_outpin)=(expected_inpin)"
+#
+# @param bel Name of a bel
+# @param expected_inpin Expected input pin of the routethrough
+# @param expected_outpin Expected output pin of the routethrough
+# @return 1 if the bel matches the expected pattern, 0 otherwise
+proc ::tincr::test_routethrough { bel expected_inpin expected_outpin } {
+    
+    set config [get_property CONFIG.EQN [get_bels $bel]]
+    
+    if {![regexp {(O[5,6])=\((A[1-6])\) ?} $config -> actual_outpin actual_inpin]} {
+       puts 0
+    }    
+    
+    if { $actual_outpin != $expected_outpin || $actual_inpin != $expected_inpin} {
+        puts "0\n$expected_inpin\n$actual_inpin\n$expected_outpin\n$actual_outpin\n"
+    }
+    
+    puts 1
+}
+
+## Test that the specified Bel is a static source bel (i.e. the configuration equation is 0 or 1).
+#
+# @param bel Name of a bel
+# @return 1 if the bel is a static source, 0 otherwise
+proc ::tincr::test_static_sources { bel } {
+    
+    set config [get_property CONFIG.EQN [get_bels $bel]]
+    
+    if {[regexp {(O[5,6])=[0,1] ?} $config -> pin] } {
+        puts 1
```

Comment:
I don't think that you can set the config.eqn of a bel (it is a read only property), but we could defintely include a `get_lut_eqn` method.